test/e2e_node/container_restart_test.go

```diff
@@ -53,11 +53,9 @@ var _ = SIGDescribe("Container Restart", feature.CriProxy, framework.WithSerial(
 			if err := resetCRIProxyInjector(e2eCriProxy); err != nil {
 				ginkgo.Skip("Skip the test since the CRI Proxy is undefined.")
 			}
```

Comment:
nit: Current approach looks like we're skipping the test when we can't reset the proxy. How about explicitly checking that the proxy is not defined, e.g.

```suggestion
                         if e2eCriProxy == nil {
                                 ginkgo.Skip("Skip the test since the CRI Proxy is undefined.")
                         }
                         framework.ExpectNoError(resetCRIProxyInjector(e2eCriProxy))
```

Reply:
/cc @pacoxu

Reply:
For current code, either is OK as resetCRIProxyInjector will return error only if the proxy is nil.

I prefer your change, more nit.

Reply:
I disagree, for happy days when error is not returned existing code resets all registered injectors. If we just return nil then we loose this.

Note: Purpose of this PR was not to refactor the tests, motivation was to resolve failures seen in pipelines and make code cleaner keeping existing functionality.